src/core/server/saved_objects/migrations/core/build_active_mappings.ts

```diff
@@ -175,6 +186,15 @@ function defaultMapping(): IndexMapping {
           },
         },
       },
+      permissions: {
+        properties: {
+          read: principals,
+          write: principals,
+          management: principals,
+          library_read: principals,
+          library_write: principals,
```

Comment:
these permissions are specific to workspace object, shall we move it to workspaces?

Or in general can we have all ACL related settings in plugin instead of defining them in `saved_object` service?

Reply:
Some functions such as `hasPermission`, `addPermission` can be moved to the workspace plugin, but the mapping change here, and the definition of the permissions object which is used by repository and saved_object_client cannot be moved to the workspace plugin.